### ISSUE TITLE
bug: Callback query string is appended after hash

### ISSUE THREAD

**Issue by tennox** [pulsejet/nextcloud-oidc-login#251]
I crafted a PR to a [oidc-client lib](https://github.com/authts/oidc-client-ts/issues/734#issuecomment-1298381823) to support parsing the URLs format that this plugin returns:
```
http://foo/#/bar?cod=param1&state=123
```

but got the reply that it's actually a bug in my Identity provider (this plugin), and correct would be:
```
http://foo/?cod=param1&state=123#/bar
```
and checked [other resources](https://stackoverflow.com/a/49833358) to confirm that this is the correct way.

According to #182 - I shall craft a PR myself, but opened this issue to give a heads-up and to ask for feedback :)

**Comment by azmeuk** [comment]
Hi @tennox
Sure please open a PR, though I wonder if this will be on the side of nextcloud-oidc-login or the underlying [OpenIDConnect-PHP](https://github.com/jumbojett/OpenID-Connect-PHP).

**Comment by pulsejet** [comment]
Yeah, I'm inclined to think the bug is either in the OIDC library or your IdP itself, depending on when this happens. I don't remember us constructing any URLs (if we do, maybe that can be removed?)
